Add detailed docstrings to model.py

nithin200417 opened the pull request:
- Added detailed docstrings to the get_linear_beta_schedule, get_cosine_beta_schedule, get_alphas, and get_alpha_bars functions.
- Added detailed docstrings to the normalization functions: normalize_to_neg_one_to_one, unnormalize_to_zero_to_one, and identity.
- Added detailed docstrings to the helper function extract.
- Added detailed docstrings to the DDIMDiffusionModel class and its methods: __init__, q_sample, p_sample, sample, and train_step.
- Improved readability and understanding of the code by providing clear explanations of the purpose, parameters, and return values of each function and method.

nithin200417 committed:
Add detailed docstrings to model.py

- Added detailed docstrings to the get_linear_beta_schedule, get_cosine_beta_schedule, get_alphas, and get_alpha_bars functions.
- Added detailed docstrings to the normalization functions: normalize_to_neg_one_to_one, unnormalize_to_zero_to_one, and identity.
- Added detailed docstrings to the helper function extract.
- Added detailed docstrings to the DDIMDiffusionModel class and its methods: __init__, q_sample, p_sample, sample, and train_step.
- Improved readability and understanding of the code by providing clear explanations of the purpose, parameters, and return values of each function and method.

singjc commented:
Hi @nithin200417, @CoderOMaster already updated the docs for model.py in #19. Can you update the docs for [model/model_interface.py](https://github.com/Roestlab/diffusion-deconvolution-dia-msms-data/blob/main/dquartic/model/model_interface.py) and [model/unet1d.py](https://github.com/Roestlab/diffusion-deconvolution-dia-msms-data/blob/main/dquartic/model/unet1d.py)

nithin200417 reviewed (commented):
Sorry for the misunderstanding i will make sure i complete the model_interface.py docstring part as per your previous message

nithin200417 committed:
Merge branch 'main' into add-docstrings

nithin200417 committed:
Added and improved docstrings for key classes and functions in unet1d.py

nithin200417 committed:
Merge branch 'add-docstrings' of https://github.com/nithin200417/diffusion-deconvolution-dia-msms-data into add-docstrings

nithin200417 reviewed (commented):
I have added and improved the docstrings for key classes and functions in the unet1d.py.Please review the changes and let me know if there are any further adjustments needed.

singjc commented:
> I have added and improved the docstrings for key classes and functions in the unet1d.py.Please review the changes and let me know if there are any further adjustments needed.

There seems to be some redundant documentation in model.py now, due to pulling int he changes that @CoderOMaster made and documented. Can you restore the model.py to the state form the main branch, unless there are parts of the documentation where you want to add more.

Can you also add documentation to [model/model_interface.py](https://github.com/Roestlab/diffusion-deconvolution-dia-msms-data/blob/main/dquartic/model/model_interface.py) as mentioned above.

nithin200417 commented:
I will make sure I do that as soon as possible

nithin200417 committed:
Update model_interface.py with improved docstrings and refactored methods

nithin200417 committed:
Restore model.py to the state from the main branch

nithin200417 commented:
Hi @singjc,

I’ve finished the following updates:

1) Restored model.py to match the main branch, removing redundant documentation.
2) Added detailed docstrings to model_interface.py for all key classes and methods.
3) Added detailed docstrings to model/unet1d.py for all key classes and methods.
Let me know if you have any feedback or if any further adjustments are needed.

Thanks!

singjc commented:
Great, everything is almost good to go. There still is an issue with the model.py file, I think you restored it to match the main branch of your fork, which is probably out of sync with the main repo's main branch. Because now the model.py is missing the documentation that was added in this commit [@00b5585
](00b55850fdea120307912d39b87c4efa129b3dbd).

So you probably need to first sync your main branch to update it, and then restore the model.py file.

nithin200417 committed:
Restore model.py to match the updated main branch

nithin200417 commented:
Hi @singjc,

I have synced my fork with the main repository, restored model.py to match the updated main branch, is this correct or is there a mistake.

singjc commented:
> Hi @singjc,
> 
> I have synced my fork with the main repository, restored model.py to match the updated main branch, is this correct or is there a mistake.

Great, everything looks good. Thanks! I will merge.
